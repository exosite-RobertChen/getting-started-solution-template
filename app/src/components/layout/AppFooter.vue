<template>
  <footer class="app-footer px-6 py-4 secondary-bg" style="width: 100vw">
    <v-row class="subtitle-1 whitevar--text">
      <v-col
        class="text-center d-flex justify-space-between px-6"
        cols="12"
        sm="4"
        lg="2"
        order-sm="1"
      >
        <a
          :href="links.info"
          target="_blank"
          whitevar--text
          data-qa="AboutLink"
          class="app-footer__link"
          >About</a
        >
        <a
          :href="links.terms"
          target="_blank"
          whitevar--text
          data-qa="TermsLink"
          class="app-footer__link"
          >Terms</a
        >
        <a
          :href="links.help"
          target="_blank"
          whitevar--text
          data-qa="HelpLink"
          class="app-footer__link"
          >Help
        </a>
      </v-col>
      <v-col cols="6" sm="4" lg="5" order-sm="0">
        &copy; {{ new Date().getFullYear() }} â€”
        {{ companyName }}
      </v-col>
      <v-col cols="6" sm="4" lg="5" class="text-right" order-sm="2">
        <div>
          v. 1.0
        </div>
      </v-col>
    </v-row>
  </footer>
</template>

<script>
export default {
  name: 'AppFooter',
  props: {
    links: {
      type: Object,
      required: true
    },
    companyName: {
      type: String,
      required: true
    }
  }
}
</script>

<style lang="scss">
.app-footer {
  &__link {
    text-decoration: none;
    transition: all 0.3s ease;
    color: var(--v-whitevar-base) !important;
    &:hover {
      text-decoration: none;
      cursor: pointer;
      color: var(--v-primary-base) !important;
    }
  }
}
</style>
