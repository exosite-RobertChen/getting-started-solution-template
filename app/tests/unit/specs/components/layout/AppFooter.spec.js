import merge from 'lodash.merge'
import { shallowComponent } from '@/../tests/unit/common'
import AppFooter from '@/components/layout/AppFooter.vue'
import branding from '@/../vendor/branding.json'

const defaultProps = {
  propsData: {
    value: true,
    companyName: branding.companyName,
    links: branding.links
  }
}

const factory = overrides =>
  shallowComponent(AppFooter, merge(defaultProps, overrides))

describe('AppFooter.vue', () => {
  const wrapper = factory()

  test('can be mounted', () => {
    expect(wrapper.contains(AppFooter)).toBe(true)
  })

  test('matches snapshot', () => {
    expect(wrapper.element).toMatchSnapshot()
  })
})
