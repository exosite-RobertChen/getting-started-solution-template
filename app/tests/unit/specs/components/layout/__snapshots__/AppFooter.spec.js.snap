// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppFooter.vue matches snapshot 1`] = `
<footer
  class="app-footer px-6 py-4 secondary-bg"
  style="width: 100vw;"
  value="true"
>
  <v-row-stub
    class="subtitle-1 whitevar--text"
    tag="div"
  >
    <v-col-stub
      class="text-center d-flex justify-space-between px-6"
      cols="12"
      lg="2"
      ordersm="1"
      sm="4"
      tag="div"
    >
      <a
        class="app-footer__link"
        data-qa="AboutLink"
        href="https://docs.exosite.com/connectivity/pdaas/device-owner/"
        target="_blank"
        whitevar--text=""
      >
        About
      </a>
       
      <a
        class="app-footer__link"
        data-qa="TermsLink"
        href="https://exosite.com/terms-conditions"
        target="_blank"
        whitevar--text=""
      >
        Terms
      </a>
       
      <a
        class="app-footer__link"
        data-qa="HelpLink"
        href="https://docs.exosite.com/connectivity/pdaas/device-owner/"
        target="_blank"
        whitevar--text=""
      >
        Help
      
      </a>
    </v-col-stub>
     
    <v-col-stub
      cols="6"
      lg="5"
      ordersm="0"
      sm="4"
      tag="div"
    >
      
      © 2020 —
      Exosite
    
    </v-col-stub>
     
    <v-col-stub
      class="text-right"
      cols="6"
      lg="5"
      ordersm="2"
      sm="4"
      tag="div"
    >
      <div>
        
        v. 1.0
      
      </div>
    </v-col-stub>
  </v-row-stub>
</footer>
`;
